### PR TITLE
Changed log.info to tidy up using fmt.sprintf

### PR DESCRIPTION
**Pull request by bharathi-tenneti**
**Description of the change:**
Modified log,Info in watches.go file to print message that doesn't necessarily state failure.

**Motivation for the change:**
The log should be info because it is not a failure but information

CLOSES: #2186

### CODE REVIEW

File: pkg/ansible/watches/watches.go [hunk 1]
```diff
@@ -275,7 +275,8 @@ func getMaxWorkers(gvk schema.GroupVersionKind, defValue int) int {
 	maxWorkers, err := strconv.Atoi(os.Getenv(envVar))
 	if err != nil {
 		// we don't care why we couldn't parse it just use default
-		log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
+		//log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
```
Comment (joelanford) [inline pkg/ansible/watches/watches.go]: No need to keep this line around anymore.

Ditto for ansibleVerbosity.

File: pkg/ansible/watches/watches.go [hunk 2]
```diff
@@ -297,7 +298,9 @@ func getAnsibleVerbosity(gvk schema.GroupVersionKind, defValue int) int {
 	))
 	ansibleVerbosity, err := strconv.Atoi(os.Getenv(envVar))
 	if err != nil {
-		log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
+		//log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
+		log.Info(fmt.Sprintf("Using default value for ansible verbosity %d", defValue))
+
```
Comment (joelanford) [inline pkg/ansible/watches/watches.go]: Nit: this extraneous empty line can be removed.

File: pkg/ansible/watches/watches.go (hunk 1 above)
Comment (camilamacedo86) [inline pkg/ansible/watches/watches.go]:
```suggestion
```

File: pkg/ansible/watches/watches.go [hunk 3]
```diff
@@ -297,7 +298,9 @@ func getAnsibleVerbosity(gvk schema.GroupVersionKind, defValue int) int {
 	))
 	ansibleVerbosity, err := strconv.Atoi(os.Getenv(envVar))
 	if err != nil {
-		log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
+		//log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
```
Comment (camilamacedo86) [inline pkg/ansible/watches/watches.go]:
```suggestion
```

File: pkg/ansible/watches/watches.go [hunk 4]
```diff
@@ -297,7 +298,9 @@ func getAnsibleVerbosity(gvk schema.GroupVersionKind, defValue int) int {
...
+		log.Info(fmt.Sprintf("Using default value for ansible verbosity %d", defValue))
```
Comment (camilamacedo86) [inline pkg/ansible/watches/watches.go]:
```suggestion
		log.Info(fmt.Sprintf("Unable to find a value for %v". Then, using default value for the ansible verbosity %d", envVar, defValue))
```
I think that is good to keep the env var to show how the user can change it.

File: pkg/ansible/watches/watches.go [hunk 5]
```diff
@@ -275,7 +275,8 @@ func getMaxWorkers(gvk schema.GroupVersionKind, defValue int) int {
 	maxWorkers, err := strconv.Atoi(os.Getenv(envVar))
 	if err != nil {
 		// we don't care why we couldn't parse it just use default
-		log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
+		//log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
+		log.Info(fmt.Sprintf("Using default value for workers %d", defValue))
```
Comment (camilamacedo86) [inline pkg/ansible/watches/watches.go]:
```suggestion
		log.Info(fmt.Sprintf("Unable to find a value for %v". Then, using default value for the workers %d", envVar, defValue))
```

**Comment by camilamacedo86** [comment]
+1 for the @joelanford suggestion. 

Just the text that I'd recommend not use the word error or failed, keep the env var for the user know how to change that and the info. So, something as: 

`Unable to find a value for %v". Then, using the default value for the environment variable %d", envVar, defValue))
`

File: CHANGELOG.md [hunk 6]
```diff
@@ -11,6 +11,7 @@
 ### Removed
 
 ### Bug Fixes
+- Changed watches.go to fix log.Info to modify message,and maxWorkers logic ([#2246](https://github.com/operator-framework/operator-sdk/pull/2246/files))
```
Comment (camilamacedo86) [inline CHANGELOG.md]: Need to be something that any user who looked that can understand. Maye something as:

> Fix log messages to make clear what are the values used and environment variables used when the operator starts.

File: go.mod [hunk 7]
```diff
@@ -16,6 +16,7 @@ require (
 	github.com/elazarl/goproxy/ext v0.0.0-20190421051319-9d40249d3c2f // indirect
 	github.com/fatih/camelcase v1.0.0 // indirect
```
Comment (camilamacedo86) [inline go.mod]: required revert the changes in the go.mod

File: pkg/ansible/watches/watches.go [hunk 8]
```diff
@@ -297,7 +294,7 @@ func getAnsibleVerbosity(gvk schema.GroupVersionKind, defValue int) int {
 	))
 	ansibleVerbosity, err := strconv.Atoi(os.Getenv(envVar))
```
Comment (joelanford) [inline pkg/ansible/watches/watches.go]: I think we should use the same logic for the ansible verbosity operator.

Since the logic is identical, I think it would make sense to define a new function with this signature:
```
func getIntegerEnvWithDefault(envVar string, defValue int) int
```
The call it for both `maxWorkers` and `ansibleVerbosity`.

File: pkg/ansible/watches/watches.go [hunk 9]
```diff
@@ -272,16 +272,13 @@ func getMaxWorkers(gvk schema.GroupVersionKind, defValue int) int {
...
-		return defValue
-	}
-
-	if maxWorkers <= 0 {
-		log.Info("Value %v not valid. Using default %v", maxWorkers, defValue)
-		return defValue
+	maxWorkers := defValue
+	if val, ok := os.LookupEnv(envVar); ok {
+		if i, err := strconv.Atoi(val); err != nil {
+			log.Info(fmt.Sprintf("Unable to find a value for %v. Using default value for maxWorkers %d", envVar, defValue))
```
Comment (joelanford) [inline pkg/ansible/watches/watches.go]: I could be wrong, but I think @camilamacedo86 was suggesting that we should log a separate message if we use the default when the environment variable is not set. I would also suggest we follow the [structured logging conventions of the logr.Logger](https://godoc.org/github.com/go-logr/logr#hdr-Parts_of_a_log_line).

So here, we can use the following since we have encountered an error trying to parse the string as an integer.
```go
log.Info("could not parse environment variable as an integer; using default value", "envVar", envVar, "default", defValue)
```

For @camilamacedo86's suggestion, if we do the `os.LookupEnv()` and `ok == false`, that means the environment variable was not set. In think in _that_ case, we could log this message:
```go
log.Info("environment variable not set; using default value", "envVar", envVar, "default", defValue)
```

@camilamacedo86 @jmrodri @bharathi-tenneti WDYT?

File: CHANGELOG.md [hunk 10]
```diff
@@ -11,6 +11,7 @@
...
```
Comment (joelanford) [inline CHANGELOG.md]: I agree with @camilamacedo86, but I think it needs to be a bit more specific and user-friendly.

Also a few nits: the change description should end in a period with the link following in parentheses, and the link should be to the base PR URL, not to the `/files` subpath.

 How about this?
```suggestion
Fixed log formatting issue that occurred while loading the configuration for Ansible-based operators. ([#2246](https://github.com/operator-framework/operator-sdk/pull/2246))
```

File: go.mod [hunk 11]
```diff
@@ -8,15 +8,9 @@ require (
 	github.com/Masterminds/semver v1.5.0 // indirect
 	github.com/Masterminds/sprig v2.22.0+incompatible // indirect
```
Comment (camilamacedo86) [inline go.mod]: Note that it has now the go.mod file and no longer the changes.

File: go.mod [hunk 12]
```diff
@@ -45,6 +45,7 @@ require (
 	github.com/ziutek/mymysql v1.5.4 // indirect
 	go.uber.org/zap v1.10.0
 	golang.org/x/tools v0.0.0-20191018212557-ed542cd5b28a
+	google.golang.org/genproto v0.0.0-20181016170114-94acd270e44e // indirect
```
Comment (camilamacedo86) [inline go.mod]: Still one change in the go.mod. Any particular reason for we add any module here?

Comment (bharathi-tenneti) [reply go.mod]: This was unintended, I have only committed watches.go file, not sure why go.mod is reflecting.

File: CHANGELOG.md [hunk 13]
```diff
@@ -15,6 +15,7 @@
 ### Removed
 
 ### Bug Fixes
+- Fixed log formatting issue that occurred while loading the configuration for Ansible-based operators. ([#2246](https://github.com/operator-framework/operator-sdk/pull/2246))
```
Comment (camilamacedo86) [inline CHANGELOG.md]: It needs to be in the unreleased section.

File: CHANGELOG.md (hunk 10 above)
Comment (camilamacedo86) [reply CHANGELOG.md]: solved

File: go.mod (hunk 12 above)
Comment (camilamacedo86) [reply go.mod]: solved

File: CHANGELOG.md (hunk 13 above)
Comment (joelanford) [reply CHANGELOG.md]: It looks like it already is.

File: pkg/ansible/watches/watches.go [hunk 14]
```diff
@@ -312,3 +299,19 @@ func getAnsibleVerbosity(gvk schema.GroupVersionKind, defValue int) int {
...
+func getIntegerEnvWithDefault(envVar string, defValue int) int {
+	val := defValue
+	if envVal, ok := os.LookupEnv(envVar); ok {
+		if i, err := strconv.Atoi(envVal); err != nil {
+			log.Info("could not parse environment variable as an integer; using default value", "envVar", envVar, "default", defValue)
+
+		} else {
+			val = i
+		}
+	} else if ok==false {
```
Comment (joelanford) [inline pkg/ansible/watches/watches.go]: Nit:
```suggestion
	} else if !ok {
```

File: CHANGELOG.md (hunk 13 above)
Comment (camilamacedo86) [reply CHANGELOG.md]: solved.

File: pkg/ansible/watches/watches.go [hunk 15]
```diff
@@ -312,3 +297,18 @@ func getAnsibleVerbosity(gvk schema.GroupVersionKind, defValue int) int {
 	}
 	return ansibleVerbosity
 }
+
+// Returns value for MaxWorkers/Ansibleverbosity based on if envVar is set or a defvalue is used.
```
Comment (joelanford) [inline pkg/ansible/watches/watches.go]: Nit: Go doc convention is to use the identifier name (in this case the function name) as the first word of the comment. Not a huge deal since this function is unexported, but a good habit nonetheless.

File: pkg/ansible/watches/watches.go [hunk 16]
```diff
@@ -312,3 +297,18 @@ func getAnsibleVerbosity(gvk schema.GroupVersionKind, defValue int) int {
...
+func getIntegerEnvWithDefault(envVar string, defValue int) int {
+	val := defValue
+	if envVal, ok := os.LookupEnv(envVar); ok {
+		if i, err := strconv.Atoi(envVal); err != nil {
+			log.Info("could not parse environment variable as an integer; using default value", "envVar", envVar, "default", defValue)
+		} else {
+			val = i
+		}
+	} else if !ok {
+ 		log.Info("environment variable not set; using default value", "envVar", envVar, "default", defValue)
```
Comment (joelanford) [inline pkg/ansible/watches/watches.go]: Looks like the sanity test is complaining that these `log.Info` messages don't start with an upper case letter.

So we need:
`Could not parse environment variable...` and `Environment variable not set...`

**Comment by joelanford** [comment]
@bharathi-tenneti Looks like a `go fmt` issue. You should be able to run `make format` to fix it.

Most editors have a Go plugin that can be configured to automatically run `go fmt` on save, which I find pretty helpful.
